exec/java-exec/src/main/java/org/apache/drill/exec/planner/ParquetPartitionDescriptor.java

```diff
@@ -125,4 +117,16 @@ private String getBaseTableLocation() {
     final FormatSelection origSelection = (FormatSelection) scanRel.getDrillTable().getSelection();
     return origSelection.getSelection().selectionRoot;
   }
+
+  @Override
+  protected void createPartitionSublists() {
+    Set<String> fileLocations = ((ParquetGroupScan) scanRel.getGroupScan()).getFileSet();
+    List<PartitionLocation> locations = new LinkedList<>();
+    for (String file: fileLocations) {
+      locations.add(new DFSPartitionLocation(MAX_NESTED_SUBDIRS, getBaseTableLocation(), file));
```

Comment:
Looks like we are still putting the file name including the directory name into heap memory, before break it into multiple sublists.  In other words, this patch will reduce the direct memory footprint allocated for value vectors. But it does not address the heap memory issue caused by very long file names, right?